vassal-app/src/main/java/VASSAL/tools/io/ZipArchive.java

```diff
@@ -425,18 +426,50 @@ private void moveFile(Path src, Path dst) throws IOException {
         // attempt to copy to the destination
         Files.copy(src, dst, StandardCopyOption.REPLACE_EXISTING);
       }
-      catch (IOException e) {
-        // copy failed
-        final String fmt = "Unable to overwrite %s, so data written to %s instead: %s";
-        throw new IOException(
-          String.format(
-            fmt,
-            dst.toAbsolutePath(),
-            src.toAbsolutePath(),
-            e.getMessage()
-          ),
-          e
-        );
+      catch (IOException copyException) {
+        Path dst2 = dst.resolveSibling(src.getFileName()); // Our "temp" name but in the destination directory
+        final String fmt = Resources.getString("Editor.ZipArchive.overwrite") + "\n%s"; //NON-NLS
+        try {
+          // Try to at least put it in same directory as the module
+          Files.copy(src, dst2, StandardCopyOption.REPLACE_EXISTING);
+
+          // Since we at least managed to copy it out of the temp directory, we delete the one out of our temp directory
+          try {
+            Files.delete(src);
+          }
+          catch (IOException ignoreDeleteException) {
+            // No action because we have a more important and informative exception to throw
+          }
+
+          // Tell where we moved the file to - don't need the obnoxious long paths in this case because they're in same directory.
+          throw new FileSystemException(
+            dst.toFile().getName(),
+            dst2.toFile().getName(),
+            String.format(
+              fmt,
+              dst.toFile().getName(),
+              dst2.toFile().getName(),
+              copyException.getMessage()
+            )
+          );
+        }
+        catch (IOException didEverythingFail) {
+          if (didEverythingFail instanceof FileSystemException) {
+            // If we picked up a FileSystemException, then it came from the inner loop meaning we at least "got halfway", so
+            // we pass that exception on rather than throwing the "worse" version.
+            throw(didEverythingFail);
```

Comment:
Shouldn't have parentheses (throw's not a function).

Reply:
done